W0212 False positive when using self.__class__

### Bug description

```python
class C:
    _X = None

    def f(self):
        if self.__class__._X is None:
            pass

self.__class__._X is clearly in the same class, so it should not complain. Instead, I get W0212.
```

### Configuration

```ini

```

### Command used

```shell
pylint /tmp/bug.py
```

### Pylint output

```python
************* Module bug
/tmp/bug.py:1:0: C0114: Missing module docstring (missing-module-docstring)
/tmp/bug.py:1:0: C0115: Missing class docstring (missing-class-docstring)
/tmp/bug.py:4:4: C0116: Missing function or method docstring (missing-function-docstring)
/tmp/bug.py:5:11: W0212: Access to a protected member _X of a client class (protected-access)
/tmp/bug.py:1:0: R0903: Too few public methods (1/2) (too-few-public-methods)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

As above, but without:
/tmp/bug.py:5:11: W0212: Access to a protected member _X of a client class (protected-access)

### Pylint version

```shell
pylint 3.3.9
astroid 3.3.8
Python 3.14.5 (main, May 11 2026, 00:00:00) [GCC 15.2.1 20260123 (Red Hat 15.2.1-7)]
```

### OS / Environment

Fedora 43

### Additional dependencies

```python

```

--- a/doc/whatsnew/fragments/11160.false_positive
+++ b/doc/whatsnew/fragments/11160.false_positive
@@ -1,0 +1,4 @@
+Fix a false positive for ``protected-access`` when a protected member is
+accessed through ``self.__class__``, which is now treated like ``type(self)``.
+
+Closes #11160

--- a/pylint/checkers/classes/class_checker.py
+++ b/pylint/checkers/classes/class_checker.py
@@ -1926,6 +1926,11 @@ a metaclass class method.",
         if self._is_type_self_call(node.expr):
             return
 
+        # If the expression is self.__class__ (or the class of any other
+        # mandatory first method parameter), that's equivalent to type(self).
+        if self._is_self_class_access(node.expr):
+            return
+
         # Check if we are inside the scope of a class or nested inner class
         inside_klass = True
         outer_klass = klass
@@ -1980,6 +1985,12 @@ a metaclass class method.",
     def _is_type_self_call(self, expr: nodes.NodeNG) -> bool:
         match expr:
             case nodes.Call(func=nodes.Name(name="type"), args=[arg]):
+                return self._is_mandatory_method_param(arg)
+        return False
+
+    def _is_self_class_access(self, expr: nodes.NodeNG) -> bool:
+        match expr:
+            case nodes.Attribute(attrname="__class__", expr=arg):
                 return self._is_mandatory_method_param(arg)
         return False
 

--- a/tests/functional/a/access/access_to_protected_members.py
+++ b/tests/functional/a/access/access_to_protected_members.py
@@ -274,3 +274,19 @@ class Issue3066:
             @staticmethod
             def _bar(i):
                 """Docstring."""
+
+
+class Issue11160:
+    """https://github.com/pylint-dev/pylint/issues/11160"""
+
+    _attr = None
+
+    def access_via_self_class(self):
+        """self.__class__ is equivalent to type(self) and should be accepted."""
+        if self.__class__._attr is None:
+            return self.__class__._attr
+        return None
+
+    def access_via_other_object_class(self, other):
+        """Accessing a protected member through another object's class warns."""
+        return other.__class__._attr  # [protected-access]

--- a/tests/functional/a/access/access_to_protected_members.txt
+++ b/tests/functional/a/access/access_to_protected_members.txt
@@ -26,3 +26,4 @@ protected-access:266:16:266:31:Issue3066.Aclass.Bclass.foobar:Access to a protec
 protected-access:267:16:267:38:Issue3066.Aclass.Bclass.foobar:Access to a protected member _attr of a client class:UNDEFINED
 protected-access:270:16:270:30:Issue3066.Aclass.Bclass.foobar:Access to a protected member _bar of a client class:UNDEFINED
 protected-access:271:16:271:37:Issue3066.Aclass.Bclass.foobar:Access to a protected member _bar of a client class:UNDEFINED
+protected-access:292:15:292:36:Issue11160.access_via_other_object_class:Access to a protected member _attr of a client class:UNDEFINED

